tests/test_hexary_trie.py

```diff
@@ -166,13 +166,20 @@ def unread_keys(self):
 def test_hexary_trie_saves_each_root():
     changes = ((b'ab', b'b'*32), (b'ac', b'c'*32), (b'ac', None), (b'ad', b'd'*32))
     expected = ((b'ab', b'b'*32), (b'ad', b'd'*32))
+
+    # track which key is expected to be present in which root
+    expected_by_root = defaultdict(set)
+    missing_by_root = defaultdict(set)
+
     db = {}
     trie = HexaryTrie(db=db)
     for key, val in changes:
         if val is None:
             del trie[key]
+            missing_by_root[trie.root_hash].add(key)
```

Comment:
Making this a `set` might be a little overboard, since the trie never goes back to an old state root right now, but it seemed easy enough to keep it general right now, in case we change the input data later.